### PR TITLE
Add status codes to test ID and Dredd output

### PR DESCRIPTION
**Pull request by antkazam**
#### :rocket: Why this change?

The primary reason for this change is that test IDs are non-unique and that makes test output hard to understand. This PR adds the `(###)` to the transaction ID so that a 200 is distinctly different from a 204, 401, 404, etc... 

```
info: Successfully connected to hooks handler. Waiting 0.1s to start testing.
pass: POST (204) /channelmapper/v1/channel duration: 25ms
skip: POST (401) /channelmapper/v1/channel
skip: POST (404) /channelmapper/v1/channel
pass: GET (200) /v1/channel/9cd4342be duration: 25ms
skip: GET (401) /v1/channel/9cd4342be
pass: GET (404) /v1/channel/9cd4342be duration: 16ms

complete: 3 passing, 0 failing, 0 errors, 3 skipped, 6 total

```

#### :memo: Related issues and Pull Requests
* Closes https://github.com/apiaryio/dredd/issues/866

#### :white_check_mark: What didn't I forget?

- [✔] To write docs
- [✔] To write tests
- [✔] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "scripts": {
     "lint": "conventional-changelog-lint --from=master && coffeelint src",
     "build": "coffee -b -c -o lib/ src/",
+    "watch": "coffee -w -b -c -o lib/ src/",
```
Comment (honzajavorek) [inline package.json]: I was reading https://github.com/apiaryio/dredd/issues/867 before this one, hence my comment there. The only important part is that I'd suggest the name of the command to be `build:watch` instead of `watch`.

File: package.json [hunk 2]
```diff
@@ -35,7 +36,7 @@
     "coffee-script": "^1.12.5",
     "colors": "^1.1.2",
     "cross-spawn": "^5.0.1",
-    "dredd-transactions": "^4.3.0",
+    "dredd-transactions": "4.2.0",
```
Comment (honzajavorek) [inline package.json]: Why this change?

Comment (antkazam) [inline package.json]: Sorry I thought I linked the issue. Currently the latest version of `dredd-transactions` has an issue introduced by one of its dependencies - https://github.com/apiaryio/dredd-transactions/issues/93

**Comment by honzajavorek** [comment]
Yes, but that's better solved by releasing a fix. One fix is already out, the other one will follow shortly.

**Comment by antkazam** [comment]
@honzajavorek I've made the changes you requested - but unfortunately the Travis CI build failed - I am trying to reproduce but having no success. The failing tests are [here](https://travis-ci.org/apiaryio/dredd/jobs/266044792) and [here](https://travis-ci.org/apiaryio/dredd/jobs/266044793) if you have any tips for how to debug.

**Comment by honzajavorek** [comment]
@antkazam I see all builds passing on the last commit. I guess you figured it out then? I'll get back shortly to do review.

**Comment by honzajavorek** [comment]
Just a nitpick from looking at the commits, you may want to fix your git or GitHub email settings to have your authorship correctly associated:

image

Only noting for your own good, this by no means blocks the PR.

**Comment by honzajavorek** [comment]
This needs to wait until https://github.com/apiaryio/dredd/issues/872 gets fixed. We need to release a fix in patch so it auto-propagates to installations the same way as did the bug. Adding a feature would close the way to auto-propagate by bumping the minor version number.

**Comment by antkazam** [comment]
Thanks @honzajavorek I was out of town until now. Really appreciate your help.

**Comment by honzajavorek** [comment]
@antkazam Thanks for the PR! 👍
